x/earn/keeper/vault_share.go

```diff
@@ -0,0 +1,80 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/earn/types"
+)
+
+// ConvertToShares converts a given amount of tokens to shares.
+func (k *Keeper) ConvertToShares(ctx sdk.Context, assets sdk.Coin) (types.VaultShare, error) {
+	totalShares, found := k.GetVaultTotalShares(ctx, assets.Denom)
+	if !found {
+		// No shares issued yet, so shares are issued 1:1
+		return types.NewVaultShare(assets.Denom, assets.Amount.ToDec()), nil
+	}
+
+	totalValue, err := k.GetVaultTotalValue(ctx, assets.Denom)
+	if err != nil {
+		return types.VaultShare{}, err
+	}
+
+	if totalValue.Amount.IsZero() {
+		return types.VaultShare{}, fmt.Errorf("total value of vault is zero")
+	}
+
+	// sharePrice := totalTokens / shareCount
+	// issuedShares = assetAmount / sharePrice
+	// issuedShares := assetAmount / (totalTokens / shareCount)
+	//               = assetAmount * (shareCount / totalTokens)
```

Comment:
very useful explanation. small suggestion: I think we should update the vars here to match the vars used in the function.

Reply:
totalTokens should be totalValue right?